docs/guidelines.md

```diff
@@ -0,0 +1,130 @@
+language features are part of "idiomatic rules" and not stylistic choices.
+Example: **Ruby**'s MRI treats variables named `_` differently.
+
+Finally there are rules that are pure preferences, even though they might be
+adopted by large bodies such as organisations and corporations. These rules
+are usually part of _competing_ standards. Exercism does not favour one over
+another. Example: **TypeScript** has a linter `tslint` (or `eslint` + plugin)
+which is maintained by a company that is not Microsoft. It competes with other
+linters such as `xo`. Most of the rules are not language features or idiomatic
+rules, and therefore stylistic choices.
+
+## Conditions and outcome
+
+For feedback generated by automated mentoring:
+
+- it **SHOULD** `disapprove_with_comment` when someone isn't using one of the
+  subjects the exercise is supposed to teach,
+- each `disapprove_with_comment` **MUST** have at least one comment,
+- each `disapprove_with_comment` **SHOULD** steer a student towards an
+  `approve_as_optimal`,
+- a student **SHOULD NOT** get a `refer_to_mentor` in a newer iteration, if
+  they've had a `disapprove_with_comment` in a previous iteration,
+- a `refer_to_mentor` **COULD** have one or more comments which will be given to
+  the mentor.
+
+There is no minimum or maximum amount of :speech_balloon: comments. However,
+each comment **SHOULD** be aiding the student towards an approvable solution.
+
+## Approvability
+
+Given the above, and to re-iterate that we focus on language fluency:
+
+### Incorrect Naming or Casing
+
+In general, if it's a _language feature_ that will be caught by a compiler or
+parser or based on official rules (which means there is a dependency on
+correctness in tools), you should `disapprove_with_comment`, preferably linking
+to the official rules. 
+
+> - :-1: disapprove if there are official guidelines on naming conventions
+> - :speech_balloon: leave a comment if there is something noteworthy
+> - :no_bell: if its a _stylistic_ preference, **and** there are _competing
+>   standards_, do not remark at all. Since there are competing standards,
+>   they're all preferences.
+> - :speech_balloon: if it's a _stylistic_ preference, **and** there is _one
+>   clear standard_, comment on it. These rules enforce idiomatic code.
+> - :question: If it's a _stylistic_ preference, **and** there is no clear
+>   standard, but most to all non-hobyist have adopted the same style, this
+>   might be idiomatic. Comment at your discretion.
+
+#### Examples
+
+- **Ruby** has a language feature where `_` is treated differently,
+  - :-1: if a student uses `_` for a variable name, but then uses it.
+- **Ruby** recognises `constants` only if they start with a **C**apital Letter,
+  - :-1: if a student uses `snake_case` for a `class` name
+- **Ruby** has _de facto_ standards on `cAsInG` and `name-ing`,
+  - :no_bell: if a student uses `camelCase` for a variable, leave it be.
```

Comment:
Why is this 🔕? That seems to contradict the next line.

Reply:
You're right. That is not the case for `Ruby`.